application/src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/system/NumItinerariesFilter.java

```diff
@@ -59,8 +61,29 @@ public List<Itinerary> flagForRemoval(List<Itinerary> itineraries) {
       itinerariesToKeep = itineraries.subList(0, maxLimit);
     }
 
-    pageCursorInputSubscriber.accept(
-      new NumItinerariesFilterResults(itinerariesToKeep, itinerariesToRemove, cropSection)
+    List<Instant> removedDepartures = itinerariesToRemove
+      .stream()
+      .map(it -> it.startTime().toInstant())
+      .toList();
+
+    Instant earliestRemovedDeparture = removedDepartures
+      .stream()
+      .min(Instant::compareTo)
+      .orElse(null);
+    Instant latestRemovedDeparture = removedDepartures
+      .stream()
+      .max(Instant::compareTo)
+      .orElse(null);
+    ItinerarySortKey pageCut = null;
+
+    if (cropSection == ListSection.HEAD) {
+      pageCut = ListUtils.first(itinerariesToKeep);
+    } else {
+      pageCut = ListUtils.last(itinerariesToKeep);
+    }
```

Comment:
So this code was moved from `NumItinerariesFilterResults` to here? I'm not sure which one is a better place for the code.

Reply:
I moved it back